database/src/postgres/rpc_server.rs

```diff
@@ -44,393 +31,548 @@ impl crate::ReaderDbManager for PostgresDBManager {
+                    AND sc.data_value IS NOT NULL
+                    and sc.account_id = sub.account_id;
+                ",
         )
-        .await?;
-
-        if let Some(active_access_keys_value) = active_access_keys {
-            let active_access_keys: std::collections::HashMap<String, Vec<u8>> =
-                serde_json::from_value(active_access_keys_value)?;
-            Ok(active_access_keys)
-        } else {
-            Ok(std::collections::HashMap::new())
+        .bind(account_id.to_string())
+        .bind(bigdecimal::BigDecimal::from(block_height))
+        .fetch(shard_id_pool.pool);
+        while let Some(row) = stream.next().await {
+            let (public_key_hex, access_key, _): (String, Vec<u8>, _) = row?;
+            let access_key_view = near_primitives::views::AccessKeyInfoView {
+                public_key: borsh::from_slice::<near_crypto::PublicKey>(&hex::decode(
+                    public_key_hex,
+                )?)?,
+                access_key: near_primitives::views::AccessKeyView::from(borsh::from_slice::<
+                    near_primitives::account::AccessKey,
+                >(
+                    &access_key
+                )?),
+            };
+            access_keys.push(access_key_view);
         }
+        Ok(access_keys)
     }
 
     async fn get_receipt_by_id(
         &self,
         receipt_id: near_indexer_primitives::CryptoHash,
-        _method_name: &str,
-    ) -> anyhow::Result<readnode_primitives::ReceiptRecord> {
-        let receipt = crate::models::ReceiptMap::get_receipt_by_id(
-            Self::get_connection(&self.pg_pool).await?,
-            receipt_id,
-        )
-        .await?;
-        readnode_primitives::ReceiptRecord::try_from((
-            receipt.receipt_id,
-            receipt.parent_transaction_hash,
-            receipt.block_height,
-            receipt.shard_id,
-        ))
-    }
-
-    async fn get_transaction_by_hash(
-        &self,
-        transaction_hash: &str,
         method_name: &str,
-    ) -> anyhow::Result<(u64, readnode_primitives::TransactionDetails)> {
-        if let Ok(transaction) = self
-            .get_indexed_transaction_by_hash(transaction_hash, method_name)
-            .await
-        {
-            Ok(transaction)
-        } else {
-            self.get_indexing_transaction_by_hash(transaction_hash, method_name)
-                .await
-        }
-    }
-
-    async fn get_indexed_transaction_by_hash(
-        &self,
-        transaction_hash: &str,
-        _method_name: &str,
-    ) -> anyhow::Result<(u64, readnode_primitives::TransactionDetails)> {
-        let (block_height, transaction_data) =
-            crate::models::TransactionDetail::get_transaction_by_hash(
-                Self::get_connection(&self.pg_pool).await?,
-                transaction_hash,
+    ) -> anyhow::Result<readnode_primitives::ReceiptRecord> {
+        let futures = self.shards_pool.iter().map(|(shard_id, pool)| {
```

Comment:
Please comment here about our chosen approach: send a query to all shards, expecting only one to result in data.